src/pages/billing/_tests/CreateAzureBillingProjectModal.test.js

```diff
@@ -0,0 +1,236 @@
+import '@testing-library/jest-dom'
+
+import { fireEvent, render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import _ from 'lodash/fp'
+import { act } from 'react-dom/test-utils'
+import { h } from 'react-hyperscript-helpers'
+import { Ajax } from 'src/libs/ajax'
+import CreateAzureBillingProjectModal from 'src/pages/billing/CreateAzureBillingProjectModal'
+import { billingProjectNameValidator } from 'src/pages/billing/List'
+import { v4 as uuid } from 'uuid'
+
+
+jest.mock('src/components/Modal', () => {
+  const { div, h } = jest.requireActual('react-hyperscript-helpers')
+  const originalModule = jest.requireActual('src/components/Modal')
+  return {
+    ...originalModule,
+    __esModule: true,
+    default: props => div({ id: 'modal-root' }, [
+      h(originalModule.default, { onAfterOpen: jest.fn(), ...props })
+    ])
+  }
+})
+jest.mock('src/libs/ajax')
+
+describe('CreateAzureBillingProjectModal', () => {
+  beforeEach(() => {
+    // Arrange
+    Ajax.mockImplementation(() => {})
+    render(h(CreateAzureBillingProjectModal, { onSuccess: jest.fn(), onDismiss: jest.fn(), billingProjectNameValidator }))
+  })
+
+  const tooShortError = 'Billing project name is too short (minimum is 6 characters)'
```

Comment:
I am not attempting to reference the message from the source files because most all of these messages are partially constructed by the validator library, and thus we do not have references to the complete messages.

Reply:
Great PR comment.